src/workerd/api/tests/streams-test.js

```diff
@@ -316,6 +316,104 @@ export const writableStreamGcTraceFinishes = {
   }
 };
 
+export const readableStreamFromAsyncGenerator = {
+  async test() {
+    async function* gen() {
+      await scheduler.wait(10);
+      yield 'hello';
+      await scheduler.wait(10);
+      yield 'world';
+    };
+    const rs = ReadableStream.from(gen());
+    const chunks = [];
+    for await (const chunk of rs) {
+      chunks.push(chunk);
+    }
+    deepStrictEqual(chunks, ['hello', 'world']);
+  }
+};
+
+export const readableStreamFromSyncGenerator = {
+  async test() {
+    const rs = ReadableStream.from(['hello', 'world']);
+    const chunks = [];
+    for await (const chunk of rs) {
+      chunks.push(chunk);
+    }
+    deepStrictEqual(chunks, ['hello', 'world']);
+  }
+};
+
+export const readableStreamFromSyncGenerator2 = {
+  async test() {
+    function* gen() {
+      yield 'hello';
+      yield 'world';
+    }
+    const rs = ReadableStream.from(gen());
+    const chunks = [];
+    for await (const chunk of rs) {
+      chunks.push(chunk);
+    }
+    deepStrictEqual(chunks, ['hello', 'world']);
+  }
+};
+
+export const readableStreamFromAsyncCanceled = {
+  async test() {
+    async function* gen() {
+      let count = 0;
+      try {
+        count++;
+        yield 'hello';
+        count++;
+        yield 'world';
+      } finally {
+        strictEqual(count, 1);
+      }
+    };
+    const rs = ReadableStream.from(gen());
+    const chunks = [];
+    for await (const chunk of rs) {
+      chunks.push(chunk);
+      return;
+    }
+    deepStrictEqual(chunks, ['hello']);
+  }
+};
+
+export const readableStreamFromThrowingAsyncGen = {
```

Comment:
Does it make sense to test the throwing sync case too?

Reply:
We could but it's going to end up exercising exactly the same code paths so I'm not sure it's actually worth it.